src/wp-includes/admin-bar.php

```diff
@@ -262,6 +262,7 @@ function wp_admin_bar_sidebar_toggle( $wp_admin_bar ) {
 function wp_admin_bar_my_account_item( $wp_admin_bar ) {
 	$user_id      = get_current_user_id();
 	$current_user = wp_get_current_user();
+	$display_name = mb_strimwidth( $current_user->display_name, 0, 40, '...' );
```

Comment:
The admin toolbar already uses `wp_html_excerpt()` to trim the Site Title (on the top level) to the first 40 characters.

```suggestion
	$display_name = wp_html_excerpt( $current_user->display_name, 40, '&hellip;' );
```